src/backend/marsha/core/models/base.py

```diff
@@ -52,18 +52,32 @@ def _get_fields_by_source_model(model):
     return fields
 
 
-class NonDeletedUniqueIndex(ConditionalUniqueIndex):
+class NonDeletedUniqueIndex(Index):
     """A special ConditionalUniqueIndex for non  deleted objects."""
 
-    condition = '"deleted" IS NULL'
+    sql_create_index = (
+        "CREATE UNIQUE INDEX %(name)s ON %(table)s%(using)s (%(columns)s)"
+        "%(extra)s%(condition)s"
+    )
 
-    def __init__(self, fields, name=None):
+    def __init__(self, *args, **kwargs):
         """Override default init to pass our predefined condition.
 
-        For the parameters, see ``ConditionalUniqueIndex.__init__``.
+        For the parameters, see ``Index.__init__``.
+
+        """
+        kwargs.update({"condition": models.Q(deleted=None)})
```

Comment:
I don't think this works. Should be:
```suggestion
        kwargs.update({"condition": models.Q(deleted__isnull=True)})
```

Reply:
It works, I have the same index created in the database.